src/sentry/static/sentry/app/views/releaseDetails.jsx

```diff
@@ -18,6 +21,54 @@ import TimeSince from 'app/components/timeSince';
 import Version from 'app/components/version';
 import withEnvironmentInQueryString from 'app/utils/withEnvironmentInQueryString';
 
+class ReleaseDetailsActions extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    release: PropTypes.object.isRequired,
+  };
+
+  handleDelete = () => {
+    let {orgId, projectId, release, api} = this.props;
+    let version = encodeURIComponent(release.version);
+    let path = `/organizations/${orgId}/releases/${version}/`;
+    let redirectPath = `/${orgId}/${projectId}/releases/`;
+    let indicator = IndicatorStore.add(t('Deleting Release...'));
+
+    api.request(path, {
+      method: 'DELETE',
+      success: () => {
+        IndicatorStore.remove(indicator);
+        browserHistory.push(redirectPath);
+      },
+      error: err => {
+        IndicatorStore.remove(indicator);
+        IndicatorStore.addError(t(err.responseJSON.detail));
```

Comment:
I would probably provide a static error message instead of using the backend error message. Also `t()` is part of our localization library that is ran (manually :() during the build process. It will only work with string literals.

Reply:
Will do.